Danki2/Assets/Scripts/AI/Behaviour/BehaviourScanner.cs

```diff
@@ -1,10 +1,12 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.Linq;
 using UnityEngine;
 
 public static class BehaviourScanner
 {
-    public static List<BehaviourData> BehaviourData { get; private set; } = new List<BehaviourData>();
+    public static List<AttributeData<BehaviourAttribute>> BehaviourData { get; private set; } = new List<AttributeData<BehaviourAttribute>>();
+    public static Dictionary<AIAction, List<AttributeData<BehaviourAttribute>>> BehaviourDataByAction { get; private set; } = new Dictionary<AIAction, List<AttributeData<BehaviourAttribute>>>();
```

Comment:
I think reintroduce BehaviourData as AttributeData<BehaviourAttribute>  (same for PlannerData).